packages/tina-cloud-next/src/index.ts

```diff
@@ -0,0 +1,62 @@
+/**
+Copyright 2021 Forestry.io Holdings, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { NextApiRequest, NextApiResponse } from 'next'
+
+export interface TinaCloudUser {
+  id: string
+  email: string
+  verified: boolean
+  role: 'admin' | 'user'
+  enabled: boolean
+  fullName: string
+}
+
+/*
+ *
+ */
+export const isAuthorized = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<TinaCloudUser | undefined> => {
+  const org = req.query.org
+  const clientID = req.query.clientID
+  const token = req.headers.authorization
+  if (
+    typeof org === 'string' &&
+    typeof clientID === 'string' &&
+    typeof token === 'string'
+  ) {
+    try {
+      // fetch identity from content server
+      const tinaCloudRes = await fetch(
+        'https://identity.tinajs.io/realm/logan/1o2sfba0u5t82qcsk16dtl7149/currentUser',
+        {
+          headers: new Headers({
+            'Content-Type': 'application/json',
+            authorization: token,
+          }),
+          method: 'GET',
+        }
+      )
+      if (tinaCloudRes.ok) {
+        const user: TinaCloudUser = await tinaCloudRes.json()
+        return user
+      }
```

Comment:
If there an error message from Tina Cloud we could also return that? would that be a good idea @jamespohalloran ?

Reply:
@mittonface What's your thoughts here? I see there is a "message" field on a failed response, but it's being set to null for unauthenticated users
<img width="196" alt="Screen Shot 2021-06-16 at 1 47 10 PM" src="https://user-images.githubusercontent.com/3323181/122260530-c4b62b80-cea9-11eb-89a7-6f1d95ab6f25.png">

Reply:
@mittonface @jamespohalloran If the token is old or malformed I get this.
```ts
{
    message: 'User is not authorized to access this resource with an explicit deny'
}
```